MeshLib/MeshEditing/Mesh2MeshPropertyInterpolation.cpp

```diff
@@ -45,71 +47,86 @@ bool Mesh2MeshPropertyInterpolation::setPropertiesForMesh(Mesh *dest_mesh, std::
         return false;
     }
 
-    GeoLib::AABB src_aabb(_src_mesh->getNodes().begin(), _src_mesh->getNodes().end());
-    GeoLib::AABB dest_aabb(dest_mesh->getNodes().begin(), dest_mesh->getNodes().end());
-    if (!src_aabb.containsAABB(dest_aabb)) {
-        ERR("MeshLib::Mesh2MeshPropertyInterpolation::setPropertiesForMesh() source mesh to small.");
-        ERR("src_aabb: %f, %f, %f | %f, %f, %f", src_aabb.getMinPoint()[0], src_aabb.getMinPoint()[1], src_aabb.getMinPoint()[2], src_aabb.getMaxPoint()[0], src_aabb.getMaxPoint()[1], src_aabb.getMaxPoint()[2]);
-        ERR("dest_aabb: %f, %f, %f | %f, %f, %f", dest_aabb.getMinPoint()[0], dest_aabb.getMinPoint()[1], dest_aabb.getMinPoint()[2], dest_aabb.getMaxPoint()[0], dest_aabb.getMaxPoint()[1], dest_aabb.getMaxPoint()[2]);
-        return false;
-    }
-
-    interpolatePropertiesForMesh(dest_mesh, dest_properties);
+    interpolatePropertiesForMesh(dest_mesh);
 
     return true;
 }
 
-void Mesh2MeshPropertyInterpolation::interpolatePropertiesForMesh(Mesh *dest_mesh, std::vector<double>& dest_properties) const
+void Mesh2MeshPropertyInterpolation::interpolatePropertiesForMesh(
+    Mesh *dest_mesh) const
 {
+    // check the existence of PropertyVector in the source mesh
+    boost::optional<MeshLib::PropertyVector<double> const&> opt_src_pv(
+        _src_mesh->getProperties().getPropertyVector<double>(_property_name));
+    if (!opt_src_pv) {
+        WARN("Did not find PropertyVector<double> \"%s\" in source mesh.",
+            _property_name.c_str());
+        return;
```

Comment:
Maybe there should be a boolean return value telling that the method didn't succeed.